File: training-data/php/exponential_search.php
<?php

function exponentialSearch(array $a, int $target): int
{
    $n = count($a);
    if ($n === 0) {
        return -1;
    }
    $bound = 1;
    while ($bound < $n && $a[$bound] < $target) {
        $bound *= 2;
    }
    [$lo, $hi] = [intdiv($bound, 2), min($bound, $n - 1)];
    while ($lo <= $hi) {
        $mid = intdiv($lo + $hi, 2);
        if ($a[$mid] === $target) {
            return $mid;
        }
        $a[$mid] < $target ? $lo = $mid + 1 : $hi = $mid - 1;
    }
    return -1;
}

$a = array_map(fn ($i) => $i * 3, range(0, 66));
echo exponentialSearch($a, 99), ' ', exponentialSearch($a, 100), ' ', exponentialSearch($a, 0), PHP_EOL;

File: training-data/go/exponential_search.go
package main

import (
	"fmt"
	"sort"
)

func exponentialSearch(a []int, target int) int {
	if len(a) == 0 {
		return -1
	}
	bound := 1
	for bound < len(a) && a[bound] < target {
		bound *= 2
	}
	lo, hi := bound/2, min(bound+1, len(a))
	i := lo + sort.SearchInts(a[lo:hi], target)
	if i < len(a) && a[i] == target {
		return i
	}
	return -1
}

func main() {
	a := make([]int, 67)
	for i := range a {
		a[i] = i * 3
	}
	fmt.Println(exponentialSearch(a, 99), exponentialSearch(a, 100), exponentialSearch(a, 0))
}
